src/compiler/checker.ts

```diff
@@ -7336,7 +7395,12 @@ namespace ts {
             if (needToCaptureLexicalThis) {
                 captureLexicalThis(node, container);
             }
-
+            if (isFunctionLike(container)) {
+                const signature = getSignatureFromDeclaration(container);

```

Comment:
Actually, `getTypeOfSymbol` won't work either. Actually, I'm unclear on when you contextually type `this`? Your writeup for the PR states that you contextually type `this` in methods in object literals, but it doesn't seem to work. In fact, I haven't been able to see any contextual typing of `this` in snippets I have written. We should discuss our strategy here.